### PR TITLE
Refactor Textsimilarity processor

### PR DESCRIPTION
**Pull request by Timoeller**
All relevant steps now reside in dataset from dicts.

Code simplified in some places.

Tests adjusted and added.

### CODE REVIEW

File: farm/data_handler/processor.py [hunk 1]
```diff
@@ -2983,173 +2934,226 @@ def file_to_dicts(self, file: str) -> [dict]:
...
-            clear_text["passages"] = positive_context + hard_negative_context
-            tokenized["passages_tokens"] = tokenized_passage
-            features["passage_input_ids"] = ctx_input_ids
-            features["passage_segment_ids"] = ctx_segment_ids
-            features["passage_attention_mask"] = ctx_padding_mask
-            features["label_ids"] = ctx_label
+                    # get tokens in string format
+                    tokenized_passage = [self.passage_tokenizer.convert_ids_to_tokens(ctx) for ctx in ctx_inputs["input_ids"]]
 
+                    if len(tokenized_passage) == 0:
```
Comment (bogdankostic) [inline farm/data_handler/processor.py:3100]: I think here we would need to iterate over tokenized_passage and check each passage, as tokenized_passage is a list of lists.

**Review by bogdankostic — APPROVED** [review]
LGTM! Just some minor comments.

File: farm/data_handler/processor.py [hunk 2]
```diff
@@ -2761,9 +2760,9 @@ def _is_impossible_to_answer_type(self, qas):
 
 class TextSimilarityProcessor(Processor):
     """
-    Used to handle the text DPR datasets that come in json format, example: nq-train.json, nq-dev.json, trivia-train.json, trivia-dev.json
-    Datasets can be downloaded from the official DPR github repository (https://github.com/facebookresearch/DPR)
+    Used to handle the Dense Passage Retrieval (DPR) datasets that come in json format, example: nq-train.json, nq-dev.json, trivia-train.json, trivia-dev.json
```
[the remark this answers is not in the document]
Comment (Timoeller) [reply farm/data_handler/processor.py]: good catch

File: farm/data_handler/processor.py [hunk 3]
```diff
@@ -2983,173 +2934,226 @@ def file_to_dicts(self, file: str) -> [dict]:
...
+                        truncation=True,
+                        padding="max_length",
+                        max_length=self.max_seq_len_passage,
+                        return_token_type_ids=True
+                    )
 
-            if len(tokenized_passage) == 0:
-                logger.warning(f"The context could not be tokenized, likely because it contains a character that the context tokenizer does not recognize")
-                return None
+                    ctx_segment_ids = np.zeros_like(ctx_inputs["token_type_ids"], dtype=np.int32)
```
[the remark this answers is not in the document]
Comment (Timoeller) [reply farm/data_handler/processor.py:3096]: I am also unhyappy about this line of code, apparently it is how the original was trained. I added a todo to check it when we train and benchmark our German DPR models

File: farm/data_handler/processor.py (hunk 1 above)
Comment (Timoeller) [reply farm/data_handler/processor.py]: tokenized strings are only for debugging purposes. So I removed the check alltogether

**Pull request merged**